### ISSUE TITLE
Unit tests written in TypeScript

### ISSUE THREAD

**Issue by claycephas** [Swiip/generator-gulp-angular#417]
This may already be possible, but I'm not sure how. Here's what I'd like to do:

- Separate interface from implementation for Angular services (i.e. `export interface IMyService` and `export class MyService`).
- Write jasmine specs in TypeScript and type service to the interface, not the class, so we get intellisense

Perhaps this will be possible after #415 is done.

**Comment by ufoscout** [comment]
Absolutely interested in this! TypeScript users would definitely love it!
Could it be extended to e2e tests too?

**Comment by JR-Utily** [comment]
I've done it manually

in script.js, duplicate the interesting part in another task

````javascript
  // specific task that will compile .spec.ts files too, but without concat neither moving them to serve/app dir
  gulp.task('scripts:test', ['scripts:app'], function(){
    return gulp.src(path.join(conf.paths.src, '/app/**/*.spec.ts'))
      .pipe($.sourcemaps.init())
      .pipe($.tslint())
      .pipe($.tslint.report('prose', { emitError: false }))
      .pipe($.typescript(tsProject)).on('error', conf.errorHandler('TypeScript'))
      .pipe($.sourcemaps.write())
      .pipe($.size())
  });


  gulp.task('scripts:app', ['tsd:install', 'constants'], function () {
  return gulp.src( [path.join(conf.paths.src, '/app/**/*.ts'), '!**/*.spec.ts'] )
    .pipe($.sourcemaps.init())
    .pipe($.tslint())
    .pipe($.tslint.report('prose', { emitError: false }))
    .pipe($.typescript(tsProject)).on('error', conf.errorHandler('TypeScript'))
    .pipe($.concat('index.module.js'))
    .pipe($.sourcemaps.write())
    .pipe(gulp.dest(path.join(conf.paths.tmp, '/serve/app')))
    .pipe(browserSync.reload({ stream: true }))
    .pipe($.size())
  });

  gulp.task('scripts', ['scripts:app']);
````
in unti-test.js, include this new task
````javascript
gulp.task('test', ['scripts:test'], function(done) {
  runTests(true, done);
});

gulp.task('test:auto', ['scripts:test', 'watch'], function(done) {
  runTests(false, done);
});
````
and you have to tweak tsd.js and tsd.json to include jasmine.t.ds as well (it wil not do it automatically as jasmine dep is not in bower but in npm)
````javascript
gulp.task('tsd:install', function () {
  var bower = require(path.join(process.cwd(), 'bower.json'));

  // reading extra dependencies from tsd.json
  var extraDependencies = require(path.join(process.cwd(), tsdJson)).extra_dependencies;

  var dependencies = [].concat(
    Object.keys(bower.dependencies),
    Object.keys(bower.devDependencies),
    extraDependencies
  );

// ....
}
````
````json
{
  "version": "v4",
  "repo": "borisyankov/DefinitelyTyped",
  "ref": "master",
  "path": ".tmp/typings",
  "bundle": ".tmp/typings/tsd.d.ts",
  "extra_dependencies": ["jasmine"]
}
````
Hope it helps !

**Comment by Swiip** [comment]
Thanks!

**Comment by JR-Utily** [comment]
De rien :)

Jérôme, de la formation que tu as donnée à Xerox en Juin

2015-08-05 16:26 GMT+02:00 Matthieu Lux <notifications@github.com>:

**Comment by zckrs** [comment]
v1.0.0-rc1 released on NPM.

[How to update my project?](https://github.com/Swiip/generator-gulp-angular/blob/master/docs/recipes/upgrade-my-current-project.md)
